test/elements/depayloader_test.exs

```diff
@@ -58,45 +54,42 @@ defmodule DepayloaderTest do
       end
     end
 
-    options = %Pipeline.Options{
-      elements: [
-        source: %Source{output: {initial_state, generator}, caps: %Packet{type: :custom_packets}},
-        depayloader: %Depayloader{packets_per_frame: 5},
-        sink: Sink
-      ]
-    }
+    structure = [
+      child(:source, %Source{output: {initial_state, generator}, stream_format: %Packet{type: :custom_packets}}),
+      child(:depayloader, %Depayloader{packets_per_frame: 5}),
+      child(:sink, Sink),
+      get_child(:source) |> get_child(:depayloader) |> get_child(:sink)
+    ]
 
-    {:ok, pipeline} = Pipeline.start_link(options)
-    Pipeline.play(pipeline)
+    pipeline = Pipeline.start_link_supervised!(structure: structure)
     assert_start_of_stream(pipeline, :sink)
 
     assert_sink_buffer(pipeline, :sink, %Buffer{payload: "Hello! How are you?"})
 
     assert_end_of_stream(pipeline, :sink)
     refute_sink_buffer(pipeline, :sink, _, 2000)
-    Pipeline.stop_and_terminate(pipeline, blocking?: true)
   end
 
   test "Depayloader should assemble the packets and form a frame" do
-    {:ok, state} = Depayloader.handle_init(%Depayloader{packets_per_frame: 5})
+    {_actions, state} = Depayloader.handle_init(nil, %Depayloader{packets_per_frame: 5})
```

Comment:
I think it would be better to match on an empty list:
```suggestion
    {[], state} = Depayloader.handle_init(nil, %Depayloader{packets_per_frame: 5})
```
In fact, the old `{:ok, state}` has became `{[], state}` and the old `{{:ok, actions}, state}` has became `{non_empty_list_of_actions, state}`

Reply:
It also applies to some other cases below